File: Restful API/Retrofit/app/src/main/java/com/okidoki/android/retrofit/dog_app/java/adapter/DogAdapter.java
package com.okidoki.android.retrofit.dog_app.java.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.okidoki.android.retrofit.R;

import java.util.List;

/**
 * Created by nikitran on 3/13/18.
 */

public class DogAdapter extends RecyclerView.Adapter<DogAdapter.ViewHolder> {
    private List<String> mDogList;
    private Context mContext;

    public DogAdapter(List<String> dogList, Context context) {
        mDogList = dogList;
        mContext = context;
    }
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_dog, parent, false);

        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(DogAdapter.ViewHolder holder, int position) {
        Glide.with(mContext)
                .load(mDogList.get(position))
                .into(holder.mDogImage);
    }

    @Override
    public int getItemCount() {
        return mDogList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder{
        ImageView mDogImage;

        public ViewHolder(View v){
            super(v);
            mDogImage = v.findViewById(R.id.dog_iv);
        }
    }
}

File: Retrofit/app/src/main/java/com/okidoki/android/retrofit/dog_app/kotlin/adapter/DogAdapter.kt
package com.okidoki.android.retrofit.dog_app.kotlin.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bumptech.glide.Glide
import com.okidoki.android.retrofit.R
import kotlinx.android.synthetic.main.item_dog.view.*

/**
 * Created by nikitran on 3/13/18.
 */
class DogAdapter(private val myDataset: List<String>, private val mContext: Context): RecyclerView.Adapter<DogAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent?.context).inflate(R.layout.item_dog, parent, false);
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        Glide.with(mContext)
                .load(myDataset.get(position))
                .into(holder.dogIv)
    }

    override fun getItemCount(): Int {
        return myDataset.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val dogIv = itemView.dog_iv
    }

}